Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ElasticWindowMultiple.py

```diff
@@ -142,54 +144,68 @@ def PyExec(self):
+            rename_alg.setProperty("OutputWorkspace", self._q_workspace)
+            rename_alg.execute()
+            rename_alg.setProperty("InputWorkspace", q2_workspaces[0])
+            rename_alg.setProperty("OutputWorkspace", self._q2_workspace)
+            rename_alg.execute()
         else:
             # Append the spectra of the first two workspaces
-            AppendSpectra(InputWorkspace1=q_workspaces[0],
-                          InputWorkspace2=q_workspaces[1],
-                          OutputWorkspace=self._q_workspace)
-            AppendSpectra(InputWorkspace1=q2_workspaces[0],
-                          InputWorkspace2=q2_workspaces[1],
-                          OutputWorkspace=self._q2_workspace)
+            append_alg = self.createChildAlgorithm("AppendSpectra", enableLogging=False)
+            append_alg.setProperty("InputWorkspace1", q_workspaces[0])
+            append_alg.setProperty("InputWorkspace2", q_workspaces[1])
+            append_alg.setProperty("OutputWorkspace", self._q_workspace)
+            append_alg.execute()
+            mtd.addOrReplace(self._q_workspace, append_alg.getProperty("OutputWorkspace").value)
+            append_alg.setProperty("InputWorkspace1", q2_workspaces[0])
+            append_alg.setProperty("InputWorkspace2", q2_workspaces[1])
+            append_alg.setProperty("OutputWorkspace", self._q2_workspace)
+            append_alg.execute()
+            mtd.addOrReplace(self._q2_workspace, append_alg.getProperty("OutputWorkspace").value)
 
             # Append to the spectra of each remaining workspace
             for idx in range(2, len(input_workspace_names)):
-                AppendSpectra(InputWorkspace1=self._q_workspace,
-                              InputWorkspace2=q_workspaces[idx],
-                              OutputWorkspace=self._q_workspace)
-                AppendSpectra(InputWorkspace1=self._q2_workspace,
-                              InputWorkspace2=q2_workspaces[idx],
-                              OutputWorkspace=self._q2_workspace)
+                append_alg.setProperty("InputWorkspace1", self._q_workspace)
+                append_alg.setProperty("InputWorkspace2", q_workspaces[idx])
+                append_alg.setProperty("OutputWorkspace", self._q_workspace)
+                append_alg.execute()
+                mtd.addOrReplace(self._q_workspace, append_alg.getProperty("OutputWorkspace").value)
+                append_alg.setProperty("InputWorkspace1", self._q2_workspace)
+                append_alg.setProperty("InputWorkspace2", q2_workspaces[idx])
+                append_alg.setProperty("OutputWorkspace", self._q2_workspace)
+                append_alg.execute()
+                mtd.addOrReplace(self._q2_workspace, append_alg.getProperty("OutputWorkspace").value)
 
             # Delete the output workspaces from the ElasticWindow algorithms
+            delete_alg = self.createChildAlgorithm("DeleteWorkspace", enableLogging=False)
```

Comment:
If we remove the `mtd.add` lines then we shouldn't need to delete the workspaces at all as they will not be visible externally.